.github/workflows/test.yml

```diff
@@ -259,35 +279,55 @@ jobs:
         with:
           path: ${{ env.BRIDGE_FILES }}
           key: bridge-files-${{ hashFiles('frostsnapp/native/src/**/*') }}
-          restore-keys: |
-            bridge-files-
 
       # Setup Java and NDK
-      - uses: actions/setup-java@v4
+      - name: Setup (Android)
+        if: matrix.build_command == 'apk'
+        uses: actions/setup-java@v4
         with:
           distribution: zulu
           java-version: 18
           cache: gradle
 
+      - name: Dependencies (Android)
+        if: matrix.build_command == 'apk'
+        run: |
+          sudo apt-get update && sudo apt-get install -y libudev-dev
       - uses: nttld/setup-ndk@v1
+        if: matrix.build_command == 'apk'
         with:
           ndk-version: r25c
 
       - uses: subosito/flutter-action@v2
         with:
           channel: stable
-          cache: true
 
       - uses: taiki-e/install-action@just
       - uses: dtolnay/rust-toolchain@stable
         with:
           targets: aarch64-linux-android,armv7-linux-androideabi,x86_64-linux-android,i686-linux-android
 
-      - name: Setup | Install Dependencies
+      - name: Setup (Linux)
+        if: matrix.os == 'ubuntu-latest' && matrix.build_command == 'linux'
         run: |
           sudo apt-get update
-          sudo apt-get install -y --no-install-recommends \
-            ninja-build libstdc++-12-dev libgtk-3-0 libgtk-3-dev libudev-dev
+          sudo apt-get install -y \
+            ninja-build \
+            libgtk-3-dev \
+            libglib2.0-dev \
+            libpango1.0-dev \
+            libgdk-pixbuf2.0-dev \
+            libcairo2-dev \
+            python3-pip \
+            git \
+            wget \
+            libudev-dev \
+            qtbase5-dev \
+            qt5-qmake \
+            qtdeclarative5-dev \
+            cmake \
+            libarchive-tools
```

Comment:
Why do we have all these dependencies now? Also wget and get? I think they are already installed. This seems to be something to do with app image stuff? (Although I don't really understand why even app images would need all these qt and libcairo etc). If so it should be done in another step. The app image just needs the flutter linux build artifact right? We should build that in one step and package the app image in a different workflow IMO.


Also can we have a `just` command to build the app image and have the CI call the just command.